server/routes/wallet.js

```diff
@@ -248,10 +248,24 @@ router.post('/public-api/buy-address', handler(async (req, res) => {
   } = req.body
   const processor = await plugins.payment
 
-  let ipAddress = req.headers[process.env.IP_HEADER_PROP_NAME] || ''
+  let ipAddress = ''
+  if (Boolean(process.env.TRUST_PROXY) && process.env.IP_HEADER_PROP_NAME) {
+    ipAddress = req.headers[process.env.IP_HEADER_PROP_NAME.toLowerCase()]
+  }
+  // first address of xff, list is comma separated
   if (ipAddress && ipAddress.indexOf(',') > -1) {
     ipAddress = ipAddress.split(',')[0]
+  } else if (ipAddress === '') {
+    ipAddress = req.ip
+  }
+  // strip the port if present.
+  const stripPort = ipAddress.split(':')
+  if (stripPort.length === 2) {
+    ipAddress = stripPort[0]
+  } else if (stripPort.length === 7) {
+    ipAddress = stripPort.pop().join(':')
```

Comment:
`stripPort.pop().join(':')` would not work. do you want to get rid from last element and then join? in this case you need something like this:
```
stripPort.pop()
ipAddress = stripPort.join(':')
```